packages/happy-dom/src/nodes/node/Node.ts

```diff
@@ -351,6 +359,60 @@ export default class Node extends EventTarget implements INode {
 		return newNode;
 	}
 
+	/**
+	 * Inserts a node to the given position.
+	 *
+	 * @param position Position to insert element.
+	 * @param element Node to insert.
+	 * @return Inserted node or null if couldn't insert.
+	 */
+	public insertAdjacentElement(position: TInsertAdjacentPositions, element: INode): INode | null {
+		if (position === 'beforebegin') {
+			if (!this.parentElement) {
+				return null;
+			}
+
+			this.parentElement.insertBefore(element, this);
+		} else if (position === 'afterbegin') {
+			this.insertBefore(element, this.firstChild);
+		} else if (position === 'beforeend') {
+			this.appendChild(element);
+		} else if (position === 'afterend') {
+			if (!this.parentElement) {
+				return null;
+			}
+
+			this.parentElement.insertBefore(element, this.nextSibling);
+		}
+
+		return element;
+	}
+
+	/**
+	 * Inserts an HTML string to the given position.
+	 *
+	 * @param position Position to insert text.
+	 * @param text HTML string to insert.
+	 * @return Inserted node or null if couldn't insert.
+	 */
+	public insertAdjacentHTML(position: TInsertAdjacentPositions, text: string): void {
+		const templateElement = <IHTMLTemplateElement>this.ownerDocument.createElement('template');
+		templateElement.innerHTML = text;
+		this.insertAdjacentElement(position, templateElement.content);
```

Comment:
```suggestion
		for(const node of XMLParser.parse(text).childNodes.slice()) {
		      this.insertAdjacentElement(position, node);
		}
```
I think we can use the XMLParser directly here :slightly_smiling_face: 

Note that you need to prepend the commit with "#228@minor:" if you apply the suggestion as well.